ytdl_nfo/nfo.py

```diff
@@ -20,7 +20,11 @@ def generate(self, raw_data):
         self.__create_child(self.top, self.data[top_name], raw_data)
 
     def __create_child(self, parent, subtree, raw_data):
-
+        # Some .info.json files may not include an upload_date.
+        if raw_data.get("upload_date") is None:
+            date = dt.datetime.fromtimestamp(raw_data["epoch"])
+            raw_data["upload_date"] = date.strftime("%Y%m%d")
```

Comment:
It seems that yt-dlp `.info.json` files do not include an `upload_date` field, as youtube-dl files do. However, both formats do include an `epoch` field, containing a number of seconds since Unix epoch. If there is no value for `upload_date`, one will be generated using the `epoch` timestamp.

Reply:
Looking at the contributor guidelines for both [youtube-dl](https://github.com/ytdl-org/youtube-dl#mandatory-and-optional-metafields) and [yt-dlp](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#mandatory-and-optional-metafields), it seems that the only required metadata fields are `id`, `title` and `url`.

The epoch property seems to be the time the tool was run/file written which may be confusing when inserted into .nfo file (e.g. in description it says 'Video Uploaded on xx/xx/xx'). I'll accept this as a temporary solution, however we should look into finding a better way of handling missing values. I've started tracking it in #7.